src/Entities/Entity.cpp

```diff
@@ -242,10 +242,17 @@ void cEntity::TakeDamage(eDamageType a_DamageType, cEntity * a_Attacker, int a_R
 {
 	TakeDamageInfo TDI;
 	TDI.DamageType = a_DamageType;
-	TDI.Attacker = a_Attacker;
+	if ((a_Attacker != nullptr) && a_Attacker->IsPawn())
+	{
+		TDI.Attacker = a_Attacker;
+	}
+	else
+	{
+		TDI.Attacker = nullptr;
+	}
```

Comment:
>We should keep the projectile as the attacker and fix whatever problem that causes.

The problem that originally caused is that wolves "attacked" arrows because they thought they were the attackers, and they never had access to the original attacker. As to why that crashes, I'm not sure, it could be wolves holding pointers to long-gone projectiles. I am currently investigating m_Target in general, it looks a bit dangerous the way it currently works.

Assuming that TDI.Attacker is always a cPawn really simplifies things. Do you have a proposal which keeps things simple while still retaining plugin compatiblity?